src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -960,32 +961,31 @@ private AnonymousContentAccessCertificate createAnonContentAccessCertificate(
     }
 
     private String createAnonPayloadAndSignature(Collection<ProductInfo> prodInfo) throws IOException {
-        List<ContentInfo> contentInfo = getContentInfo(prodInfo);
-        List<org.candlepin.model.Content> contents = convertContentInfoToContent(contentInfo);
-        Map<org.candlepin.model.Content, Boolean> activeContent = new HashMap<>();
-        contents.forEach(content -> activeContent.put(content, true));
+        Map<ContentInfo, Boolean> contentInfo = getContentInfo(prodInfo);
+        Map<org.candlepin.model.Content, Boolean> activeContent = convertContentInfoToContent(contentInfo);
         PromotedContent promotedContent = new PromotedContent(new AnonymousContentPrefix());
         byte[] data = createContentAccessDataPayload(null, activeContent, promotedContent);
 
         return createPayloadAndSignature(data);
     }
 
     /**
-     * Retrieves all of the {@link ContentInfo} from the provided {@link ProductInfo}
+     * Retrieves all of the {@link ContentInfo} and their enabled/disabled status from the provided
+     * {@link ProductInfo}
      *
      * @param prodInfo
      *  the product info that contains content info
      *
-     * @return all of the content info from the provided product info
+     * @return a map that contains all the content info and their enabled/disabled status based on the
+     * provided product info
      */
-    private List<ContentInfo> getContentInfo(Collection<ProductInfo> prodInfo) {
+    private Map<ContentInfo, Boolean> getContentInfo(Collection<ProductInfo> prodInfo) {
         return prodInfo.stream()
             .filter(Objects::nonNull)
             .map(ProductInfo::getProductContent)
             .flatMap(Collection::stream)
             .filter(Objects::nonNull)
-            .map(ProductContentInfo::getContent)
-            .toList();
+            .collect(Collectors.toMap(ProductContentInfo::getContent, ProductContentInfo::isEnabled));
```

Comment:
I think this could lead to some inconsistency in cases where the input collection has multiple products with the same content but having different enabled states -- the order may not be consistent from run to run.

We may need to fallback to the `enabled ||= pcinfo.isEnabled` behavior rather than just taking it straight up.